src/tooltip/tooltip.js

```diff
@@ -284,15 +280,16 @@ angular.module( 'ui.bootstrap.tooltip', [ 'ui.bootstrap.position', 'ui.bootstrap
             /**
              * Observe the relevant attributes.
              */
-            if (!options.useContentExp) {
-              attrs.$observe( type, function ( val ) {
+            attrs.$observe( type, function ( val ) {
+              $timeout(positionTooltip, 0, false);
+              if (!options.useContentExp) {
```

Comment:
I understand what you're saying. I'm explaining to you that if `options.useContextExp` is true, the tooltip's value will never change. That's why the check was there, this observer isn't needed if that check returns true. It also means you don't need to observe the tooltip's value if the the `options.useContextExp`.

I'm not saying to separate the repositionTooltip logic out of this observer, I'm saying that the check should still be outside. You shouldn't need to observe the tooltip's value at all if `options.useContextExp` is true because it'll never change..